integration/admin-client/src/main/java/org/keycloak/admin/client/resource/GroupResource.java

```diff
@@ -82,7 +82,7 @@
      */
     @PUT
     @Consumes(MediaType.APPLICATION_JSON)
-    void update(GroupRepresentation rep);
+    Response update(GroupRepresentation rep);
```

Comment:
I am not sure about changing this to "Response" . My vote is to keep return type to "void" .

With the "void", it means that exception is thrown for the client if something is bad (duplicate name etc). Which is in general better IMO as client knows that something was wrong. With "Response", the client explicitly needs to check if response status was correct, also he may need to close the responses etc. Just my 2 cents.

@stianst wdyt about this?

BTV. It is fine to change the server method with return type "Response" and just the client keep to "void" . It should work just fine.

Reply:
@mposolda @stianst We haven't been consistent about this.  I was trying to follow convention of other API's.  Now I see that some allow the exception to be thrown while others stuff it into a Response.

I would prefer to not change this just because I would have to go in and rewrite a lot of stuff, especially the tests.  But it's OK if you guys think that's what I should do.

Reply:
I can see that admin-client returns usually "void" on all the places where @PUT method is used. There is no other @PUT method, which returns "Response" . Also changing org.keycloak.admin.client.resource.GroupResource.update to return "Response" instead of "void" can theoretically break backwards compatibility for some clients, who use this (they rely on catching the Exception and now they will need to change to rely on checking Response status).

Also I think that admin client is unrelated to the admin UI as admin UI doesn't use admin-client? IMO it's not an issue to keep just org.keycloak.admin.client.resource.GroupResource.update to return "void", but return "Response" from the server-side org.keycloak.services.resources.admin.GroupResource . Both GroupResource class from admin-client and keycloak-services are unrelated to each other - the only contract between them is REST interface, not tightly coupled java interface or something.

Reply:
Yes, I was talking about the REST API and not the client.  I guess it's not so bad if we only need to leave the client as returning void.

Reply:
I was talking about the REST API.  I guess it's not so bad if we only need to change the client signature.